txcache/selectionTracker.go

```diff
@@ -436,6 +442,7 @@ func (st *selectionTracker) deriveVirtualSelectionSession(
 			"latestRootHash", st.latestRootHash,
 			"session rootHash", rootHash,
 		)
+		return nil, errRootHashMismatch
```

Comment:
[question]
Is this backwards compatible? Or is this just new code for supernova?

Reply:
should also apply on legacy, as the onExecuted is done there as well.
If it is not done everywhere, should be added, and if added in all needed places, then it should be backwards compatible